astropy/nddata/ccddata.py

```diff
@@ -0,0 +1,568 @@
+        # the primary header is empty.
+        if hdu == 0 and hdus[hdu].data is None:
+            for i in range(len(hdus)):
+                if hdus.fileinfo(i)['datSpan'] > 0:
+                    hdu = i
+                    hdr = hdr + hdus[hdu].header
+                    log.info("first HDU with data is extension "
+                             "{0}.".format(hdu))
+                    break
+
+        if 'bunit' in hdr:
+            fits_unit_string = hdr['bunit']
+            # patch to handle FITS files using ADU for the unit instead of the
+            # standard version of 'adu'
+            if fits_unit_string.strip().lower() == 'adu':
+                fits_unit_string = fits_unit_string.lower()
+        else:
+            fits_unit_string = None
+
+        if unit is not None and fits_unit_string:
+            log.info("using the unit {0} passed to the FITS reader instead of "
+                     "the unit {1} in the FITS file.".format(unit,
+                                                             fits_unit_string))
+
+        use_unit = unit or fits_unit_string
+        # Try constructing a WCS object. This may generate a warning, but never
+        # an error.
+        wcs = WCS(hdr)
+        # Test for success by checking to see if the wcs ctype has a non-empty
+        # value.
+        wcs = wcs if wcs.wcs.ctype[0] else None
+        ccd_data = CCDData(hdus[hdu].data, meta=hdr, unit=use_unit,
+                           mask=mask, uncertainty=uncertainty, wcs=wcs)
+
+    return ccd_data
+
+
+def fits_ccddata_writer(ccd_data, filename, hdu_mask='MASK',
+                        hdu_uncertainty='UNCERT', hdu_flags=None, **kwd):
+    """
+    Write CCDData object to FITS file.
+
+    Parameters
+    ----------
+    filename : str
+        Name of file.
+
+    hdu_mask, hdu_uncertainty, hdu_flags : str or None, optional
+        If it is a string append this attribute to the HDUList as
+        `~astropy.io.fits.ImageHDU` with the string as extension name.
+        Flags are not supported at this time. If ``None`` this attribute
+        is not appended.
+        Default is ``'MASK'`` for mask, ``'UNCERT'`` for uncertainty and
+        ``None`` for flags.
+
+    kwd :
+        All additional keywords are passed to :py:mod:`astropy.io.fits`
+
+    Raises
+    -------
+    ValueError
+        - If ``self.mask`` is set but not a `numpy.ndarray`.
+        - If ``self.uncertainty`` is set but not a
+          `~astropy.nddata.StdDevUncertainty`.
+        - If ``self.uncertainty`` is set but has another unit then
+          ``self.data``.
+
+    NotImplementedError
+        Saving flags is not supported.
+    """
+    hdu = ccd_data.to_hdu(hdu_mask=hdu_mask, hdu_uncertainty=hdu_uncertainty,
+                          hdu_flags=hdu_flags)
+    hdu.writeto(filename, **kwd)
+
+
+# This should be be a tuple to ensure it isn't inadvertently changed
+# elsewhere.
+_recognized_fits_file_extensions = ('fit', 'fits', 'fts')
```

Comment:
I think this could be removed now. `ImageFileCollection` will use the extensions provided in `ccdproc` and the reader shouldn't need them (because the `fts` has been added to the supported extensions).